### DISCUSSION TITLE
Is there a builtin way in regex to get the "depth" of a capture ?

### DISCUSSION THREAD

**Discussion by NovaliX-Dev** [rust-lang/regex#1239] [Q&A] [answered]
By depth i mean by how many other captures a specific one is surrounded.

For example : 
- `(bla(?<a>[[:alpha:]]*))` : Here the capture `a` will have the depth 2
- `(bla)` : this unnamed capture will have a depth 1

Is this a feature already implemented in regex ? Or should i make a feature request for it ?

**Comment by BurntSushi** [comment, marked as answer]
You can do it manually by getting an `Hir` from `regex-syntax` and computing the depth yourself.

I don't see a reason to provide anything more convenient in the API.
